### PR TITLE
Add build instructions for developers

### PR DESCRIPTION
**Pull request by smatting**
This PR
- adds documentation for how to build wire-server to `./docs/developer/building.md` (and link from README.md in project root)
- includes `hie.yaml` in repo (so it doesn't have to be generated)
- add makefile target for `cabal.project.local` file
- update `delete-cache-on-linker-errors` makefile target: delete `~/.cabal/store` instead of `~/.cabal`

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] If this PR changes development workflow or dependencies, they have been A) automated and B) documented under docs/developer/. All efforts have been taken to minimize development setup breakage or slowdown for co-workers.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: docs/developer/building.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+
+1. Create a `.envrc.local` file with these contents
+
+    ```
+    export COMPILE_NGINX_USING_NIX=1
+    export WIRE_BUILD_WITH_CABAL=1
+    ```
+
+   and reload the direnv via `direnv reload`
```
Comment (smatting) [inline docs/developer/building.md:15]: I'd like to remove these variables in a separate PR

**Review by jschaul — APPROVED** [review]
Thanks! Looks good! Note that this may conflict with #2623 in that one makefile target so a rebase might be needed if that other PR is merged first.

**Review by fisx — APPROVED** [review]
This also conflicts with https://github.com/wireapp/wire-server/pull/2622, but it should be easy to fix: just rebase whoever goes second over the other, `git mv docs/developer/building.md docs/src/developer/developer/building.md`, and add a toc entry to `docs/src/developer/index.rst`.

**Comment by smatting** [comment]
> Thanks! Looks good! Note that this may conflict with #2623 in that one makefile target so a rebase might be needed if that other PR is merged first.

I removed the duplicated changes from this PR.

**Pull request merged**